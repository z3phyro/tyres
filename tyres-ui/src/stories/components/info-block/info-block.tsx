import { JSX } from "solid-js";
import InfoIcon from "../icons/info.icon";

export interface InfoBlockProps {
  title: string;
  children: JSX.Element | string;
}
export function InfoBlock(props: InfoBlockProps) {
  return (
    <article class="text-gray-400 text-sm text-light">
      <h4 class="flex items-center gap-2 text-xs">
        <InfoIcon class="text-sm" /> {props.title}
      </h4>
      <p class="pl-[22px]">{props.children}</p>
    </article>
  );
}
