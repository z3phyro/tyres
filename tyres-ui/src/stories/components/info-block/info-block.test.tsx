import { test, expect } from "vitest";
import { render } from "@solidjs/testing-library";
import { InfoBlock } from "./info-block";

test("InfoBlock renders", () => {
  const { getByRole } = render(() => <InfoBlock />);

  const infoBlock = getByRole("article") as HTMLElement;
  expect(infoBlock).toBeTruthy();
});

test("InfoBlock renders with custom content", () => {
  const { getByRole } = render(() => <InfoBlock>InfoBlock content</InfoBlock>);

  const infoBlock = getByRole("article");
  expect(infoBlock.lastChild?.textContent).toBe("InfoBlock content");
});

test("Infoblock snapshot test", () => {
  const { asFragment } = render(() => <InfoBlock>InfoBlock content</InfoBlock>);

  expect(asFragment()).toMatchSnapshot();
});

