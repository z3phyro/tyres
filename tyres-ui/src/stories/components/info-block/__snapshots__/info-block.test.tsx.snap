// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Infoblock snapshot test 1`] = `"<article class="text-gray-400 text-sm text-light"><h4 class="flex items-center gap-2 text-xs"><svg class="text-sm" fill="none" role="img" stroke-width="2" xmlns="http://www.w3.org/2000/svg" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" viewBox="0 0 24 24" height="1em" width="1em" style="overflow: visible; color: currentcolor;"><path d="M12 2A10 10 0 1 0 12 22 10 10 0 1 0 12 2z"></path><path d="M12 16 12 12"></path><path d="M12 8 12.01 8"></path></svg> </h4><p class="pl-[22px]">InfoBlock content</p></article>"`;
